exotica_core/src/visualization_meshcat.cpp

```diff
@@ -89,6 +90,16 @@ void VisualizationMeshcat::Initialize(bool use_mesh_materials)
     // https://github.com/rdeits/meshcat-python/blob/aa3865143120f5ace8e62aab71d825e33674d277/src/meshcat/visualizer.py#L60
     ConnectZMQ();
     web_url_ = RequestWebURL();
+    if (file_url_ == "")
+    {
+        std::regex url_regex("(.*):(?:\\d+)(?:\\/static\\/)");
+        std::smatch match;
+        if (std::regex_search(web_url_, match, url_regex) && match.size() > 1)
+        {
+            file_url_ = match.str(1) + ":9000/files/";
```

Comment:
This is hard-coded while the file server allows for it being configurable.

Reply:
This is just default behaviour. You can specify the full URL including the port to override it.